rust_snuba/src/factory.rs

```diff
@@ -156,3 +177,53 @@ impl ProcessingStrategyFactory<KafkaPayload> for ConsumerStrategyFactory {
         }
     }
 }
+
+#[derive(Clone)]
+struct SchemaValidator {
+    schema: Option<Arc<Schema>>,
+    enforce_schema: bool,
+}
+
+impl SchemaValidator {
+    async fn process_message(
```

Comment:
A lot of this code is duplicated from the rust processor, is it possible to share some of the implementation?

Reply:
Per message.